src/sbl_filing_api/entities/repos/submission_repo.py

```diff
@@ -114,6 +117,22 @@ async def update_submission(submission: SubmissionDAO, incoming_session: AsyncSe
     return await upsert_helper(session, submission, SubmissionDAO)
 
 
+async def check_expired_submissions():
+    session = SessionLocal()
+    check_states = [
+        SubmissionState.SUBMISSION_STARTED.value,
+        SubmissionState.SUBMISSION_UPLOADED.value,
+        SubmissionState.VALIDATION_IN_PROGRESS.value,
+    ]
+    stmt = select(SubmissionDAO).filter(SubmissionDAO.state.in_(check_states))
+    submissions = (await session.scalars(stmt)).all()
+    for s in submissions:
+        if abs(s.submission_time.timestamp() - datetime.now().timestamp()) > settings.expired_submission_diff_secs:
```

Comment:
Also, can we add a `WARN`-level log statement in here when this happens?

Reply:
Yup to both.

Also yeah definitely, wasn’t considering multiple pods.  What I’ve done in the past is a single stateful set pod (you can singleton em) that handle this sort of thing.  It would just be the scheduled task and would call an endpoint on the service (which could then be handled by any of the api pods)

There wasn’t any sort of timeout I could find to set on the background task itself.  However talking with ye olde chat we could use a mixture of the background task, a wait_for task with timeout and gather them.  The wait task timeout error could check, and then set, the sub state if it’s not in a good validation state.  Problem with that approach is it would only handle situations where the sub actually entered into the validation flow.  If someone failed outside of that (during  S3 for example) that we for some reason don’t handle gracefully prior to the background task creation, we’d miss setting the expired state. Just a consideration to tying it to the background task